### PR TITLE
Improve brimcap error handling

### PR DESCRIPTION
**Pull request by nwt**
Zui displays a Node exception if brimcap exits without reading its standard input to EOF.  Display an error toast with details instead.

Update, here's how the branch looks now.

https://github.com/brimdata/zui/assets/3460638/5d6de8d6-535b-4ecb-bc07-8276d9c6cc9b

### CODE REVIEW

**Comment by philrz** [comment]
I tested out the branch with the pcap from https://github.com/brimdata/zui/issues/2926#issuecomment-1879898700. The error presentation is improved enough that I'd consider this a fix for #2926.

image

File: apps/zui/src/util/error-to-string.ts [hunk 1]
```diff
@@ -15,6 +15,9 @@ export function errorToString(e: unknown) {
     if ("error" in e && typeof e.error == "string") {
       return e.error
     } else if ("message" in e && typeof e.message === "string") {
+      if ("details" in e && Array.isArray(e.details)) {
+        return [e.message, ...e.details].join("\n")
```
Comment (nwt) [inline apps/zui/src/util/error-to-string.ts:19]: @jameskerr: react-hot-toast ignores these newlines (as well as any embedded in the strings in `e.details`) so these notifications aren't always as readable as they could be, as you can see here. Got any ideas for how we can improve on this?

image

**Comment by philrz** [comment]
I'm currently testing this branch at commit 23af3d0.

The attached video shows we're now persisting the error that bubbled up from Brimcap and it's well-formatted, which is great. However, a JavaScript "Uncaught Exception" and stack dump still pop up first and that needs to be dismissed, which was unexpected. I made a Dev build just to double check that this wasn't one of those "it will show up only in DevTools in a real build" situations, but it appeared there too.

https://github.com/brimdata/zui/assets/5934157/b8f76d67-968d-4a93-98af-9d1b9a401100

**Comment by philrz** [comment]
@jameskerr: The e2e test looks good. FYI, I made a temporary branch based on this branch to which I merged `main` so I could have the knobs available in my improved `e2e.yml`. With that I did this [Actions run](https://github.com/brimdata/zui/actions/runs/7617421001) that shows your e2e test running reliably on all OSes in CI.

At this point I think I'm good with this one merging, but since @nwt was already involved in this code and I know the Zui/Brimcap error channel has some complexities, I'd feel better if he had a chance to give it a 👍 first. I guess since he opened the PR he technically can't be the one to click "Approve" but I'll wait for him to give the word somewhere and then I'll do the clicking.

**Review by philrz — APPROVED** [review]
@nwt and I talked offline and he didn't feel he had more to add to the review, so I'm hereby giving it the 👍.

**Pull request merged**